### ISSUE TITLE
Timeouts after downloading a lot of files (iOS)

### ISSUE THREAD

**Issue by elvisgraho** [apache/cordova-plugin-file-transfer#260]
# Bug Report

## Problem
When you download a lot (more than 200) of files (images) consecutively, NSMutableURLRequest gets stuck after 50+ successful requests (and a few requests after that), because of the connection timeout.

You can see when it starts to happen, because few connections will get slower and slower until eventually it will time out.

Possibly this [link](https://stackoverflow.com/questions/13369386/how-to-cancel-a-persistent-connection-using-nsurlconnection) is related, idk.

As mentioned in the link above, [connection cancel] does nothing.

### What is expected to happen?

No timeouts.

### What does actually happen?

Timeouts.

Rewriting success call did not help but seem to improve a performance a little bit.

``` objective-c
    // remove connection for activeTransfers
    @synchronized (command.activeTransfers) {
        [connection cancel];
        [command.activeTransfers removeObjectForKey:objectId];
        // remove background id task in case our upload was done in the background
        [[UIApplication sharedApplication] endBackgroundTask:self.backgroundTaskID];
        self.backgroundTaskID = UIBackgroundTaskInvalid;
        
        [self.command.commandDelegate sendPluginResult:result callbackId:callbackId];
    }
```

*Note that my JS code explicitly waits for ```[self.command.commandDelegate sendPluginResult:result callbackId:callbackId];``` before sending next request.

**Comment by sdkester** [comment]
@elvisgraho We removed the cached response to improve the memory usage when transferring a lot of files.

```objc
CDVFileTransferDelegate* currentDelegate = command.activeTransfers[objectId];
[[NSURLCache sharedURLCache] removeCachedResponseForRequest:currentDelegate.connection.currentRequest];
```

**Comment by nunohorta** [comment]
@elvisgraho are you doing multiple downloads at the same time?

@sdkester did you add that code after each transfer? I'm trying to workaround timeouts here too and I keep seeing errors about "Background task X, was created over 30 seconds ago..." so I think iOS is killing the tasks which then results in a file transfer timeoutt

**Comment by elvisgraho** [comment]
@nunohorta No. One download after another. _"*Note that my JS code explicitly waits for [self.command.commandDelegate sendPluginResult:result callbackId:callbackId]; before sending next request."_

"Background task X, was created over 30 seconds ago" issue is fixable when you remove the concurrency in the Objective-C request code. 

After that I'v got further Problems that are deeply rooted in iOS and deprecated HTTP client functions that are used in this plugin. The issue is that "[connection cancel];" does not actually cancel the connection. It is an asynchronous process that tells the low-level code to perform a cleanup.

What fixed the issue for me is: 
1. Ditch this plugin
2. Use HTTP from @ionic-native/http (You can download files with it.)

**Comment by sdkester** [comment]
@nunohorta Like @elvisgraho I control the number of file transfers that are sent to the cordova plugin. My preferred library is the [queue flow control in AsyncJS](https://caolan.github.io/async/v3/docs.html#queue). This lets us send one or two file download requests at a time with a realtime queue waiting for additions and completions.
